### PR TITLE
cmd: support --config option

### PR DESCRIPTION
**Pull request by liuming50**
Let's support --config option by setting environment variable DOCKER_CONFIG instead of ignoring it for docker compatibility.

Add test case for config option.

#### Does this PR introduce a user-facing change?

```release-note
pdoman run --config option is now supported.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @liuming50 
LGTM, other then fixing the lint errors.
@containers/podman-maintainers PTAL

**Comment by liuming50** [comment]
Thanks @rhatdan for reviewing.

Changes in patch set 2:

1 Shore up git message.

2 Fix a comment typo in the test case.

**Comment by baude** [comment]
what config are you exactly implying here?

**Review by vrothberg — COMMENTED** [review]
I think we should keep doing what we're doing.  Podman does not support Docker config, so I prefer the warning rather than giving the illusion of --config being properly supported.

**Comment by liuming50** [comment]
> what config are you exactly implying here?

Hi, @baude, basically to support using podman's docker wrapper like this: sudo docker --config /etc/docker login

**Comment by vrothberg** [comment]
> @vrothberg but podman does support DOCKER_CONFIG env variable right now, even though it's not properly supported neither? Wouldn't this be a same case?

That's a fair point.  The DOCKER_CONFIG is used for authentication purposes.  If `--config` is documented as such I won't complain.

**Comment by liuming50** [comment]
> > @vrothberg but podman does support DOCKER_CONFIG env variable right now, even though it's not properly supported neither? Wouldn't this be a same case?
> 
> That's a fair point. The DOCKER_CONFIG is used for authentication purposes. If `--config` is documented as such I won't complain.

@vrothberg good point, I will send a V3 documenting it.

**Comment by liuming50** [comment]
Changes in patch set V3:

1 Rebased to the latest commit of main branch.

2 Modified commit message and help function, to state that this --config only for locating authentication file.

**Comment by mheon** [comment]
Changes LGTM, but linter is complaining.

**Comment by liuming50** [comment]
Changes in patch set V4:

1 Rebased on the latest commit of main branch.

2 Fixed a invalid "fmt.Fprintf" usage that causing linter complaining.

**Comment by rhatdan** [comment]
/approve
/lgtm
/hold
Thanks @liuming50

**Comment by liuming50** [comment]
@rhatdan I saw it still failed on some tests due to missing content in docs, so I made this V5, by adding "--config" description in docs/source/markdown/podman.1.md

File: docs/source/markdown/podman.1.md
Comment (liuming50) [reply docs/source/markdown/podman.1.md:36]: @vorburger good point, will do that in V6

**Comment by liuming50** [comment]
Changes in patch set V7:

1 Add description in docs/source/markdown/podman.1.md, explain that only authentication parts are supported by "--config", as suggested by @vrothberg

**Review by vrothberg — APPROVED** [review]
LGTM, thanks!

**Comment by rhatdan** [comment]
Looks like --config needs a shell completion argument.

**Comment by liuming50** [comment]
> Looks like --config needs a shell completion argument.

@rhatdan Thanks for the review, I have pushed a V8 to fix that, verified bash completion works on a qemuarm64 machine.

**Comment by liuming50** [comment]
Changes in patch set V9:

1 Fixed a wrong order of global "--config" in test script.

**Comment by rhatdan** [comment]
Looks like your test is failing.

**Comment by liuming50** [comment]
> Looks like your test is failing.

Yes, I am also aware of that, strange, I did not see why it's failing, the test passed if I run it manually with my docker hub account. I have resent the MR, see how it's going.

File: test/system/150-login.bats
Comment (edsantiago) [inline test/system/150-login.bats:103]: Suggestion:
```diff
     run_podman ... login \
+        --tls-verify=false \
         --username ...
```

Comment (liuming50) [reply test/system/150-login.bats]: @edsantiago thanks! Have made a new MR adding that part in

**Pull request merged**
